src/document.rs

```diff
@@ -1,27 +1,98 @@
 use serde::{Deserialize, Serialize};
+use std::fmt::Debug;
 
+/// Options for document insertion.
 #[derive(Serialize, Deserialize, Debug)]
-pub struct Document<T> {
-    #[serde(skip_serializing_if = "String::is_empty")]
-    _id: String,
+#[serde(rename_all = "camelCase")]
+pub struct DocumentInsertOptions {
+    /// Wait until document has been synced to disk.
+    pub wait_for_sync: Option<bool>,
+    /// Additionally return the complete new document under the attribute new in the result.
+    pub return_new: Option<bool>,
+    /// Additionally return the complete old document under the attribute old in the result. Only available if the overwrite option is used.
+    pub return_old: Option<bool>,
+    /// If set to true, an empty object will be returned as response.
+    /// No meta-data will be returned for the created document.
+    /// This option can be used to save some network traffic.
+    pub silent: Option<bool>,
+    /// If set to true, the insert becomes a replace-insert.
+    /// If a document with the same _key already exists the new document is not rejected with unique constraint violated but will replace the old document.
+    pub overwrite: Option<bool>,
+    /// TODO add nice formatted documentation from official doc
+    #[cfg(arango3_7)]
```

Comment:
You might use cfg in the wrong way.  It should be `#[cfg(feature = "arango3_7")]`.

Reply:
I tried to change to #[cfg(feature = "arango3_7")], but I get the same warning in my IDE. Also we did write stuff this way on CollectionDetails for example

```
#[derive(Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CollectionDetails {
    pub status_string: String,
    pub key_options: CollectionKeyOptions,
    pub wait_for_sync: bool,
    pub write_concern: u16,
    #[cfg(rocksdb)]
    pub cache_enabled: bool,
    #[cfg(rocksdb)]
    pub object_id: String,
    #[cfg(mmfiles)]
    pub is_volatile: bool,
    #[cfg(mmfiles)]
    pub do_compact: bool,
    #[cfg(mmfiles)]
    pub journal_size: usize,
    #[cfg(mmfiles)]
```
Anyway, it is just a warning from the IDE, because it does compile as expected :)

Reply:
Well, you are right! The two way to use cfg flag both work. Thank you.